Sources/YipYipSwift/Classes/View/PagedTableView/YipYipPagedTableViewHelper.swift

```diff
@@ -0,0 +1,114 @@
+//
+//  PagedTableViewHelper.swift
+//  Pods-YipYipSwiftExample
+//
+//  Created by Rens Wijnmalen on 01/07/2020.
+//
+
+import UIKit
+
+public class YipYipPagedTableViewHelper {
+    
+    private let loadMargin: Int
+    private var tableView: YipYipActivityStatusTableView
+    private var viewModel: YipYipPagedTableViewModel
+    private var isLoadingNextPage: Bool = false
+    
+    // ----------------------------------------------------
+    // MARK: - Initializers
+    // ----------------------------------------------------
+    
+    public init(_ tableView: YipYipActivityStatusTableView, viewModel: YipYipPagedTableViewModel, autoLoadFirstItems: Bool = true, loadMargin: Int = 5) {
+        self.tableView = tableView
+        self.viewModel = viewModel
+        self.loadMargin = loadMargin
+        
+        if autoLoadFirstItems {
+            self.scrollViewDidScroll(self.tableView)
+        }
+    }
+    
+    // ----------------------------------------------------
+    // MARK: - Public methods
+    // ----------------------------------------------------
+    
+    public func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        if !self.isLoadingNextPage, !self.viewModel.allItemsLoaded {
+            if let visibleIndexPaths = self.tableView.indexPathsForVisibleRows, let lastVisibleIndexPaths = visibleIndexPaths.last {
+                let numberOfItems = self.viewModel.numberOfItemsForSection(0)
+                if (lastVisibleIndexPaths.row >= (numberOfItems - self.loadMargin)) {
+                    self.loadNextPage()
+                }
+            }
+            else {
+                self.loadNextPage()
+            }
+        }
+    }
+    
+    // ----------------------------------------------------
+    // MARK: - Private methods
+    // ----------------------------------------------------
+    
+    private func loadNextPage() {
+        self.isLoadingNextPage = true
```

Comment:
Moet hier niet nog worden afgevangen of ie al een pagina aan het laden is? Ik geloof dat het niet zeker is dat antwoorden van http requests altijd in dezelfde volgorde aankomen als de request, dus als je meerdere pages vlak na elkaar gaat opvragen heb je kans dat ze in de verkeerde volgorde aankomen. (Dat hoeft op zich geen probleem te zijn als de DataSource daar maatregelen tegen neemt, maar het is fijn als een dit soort functionaliteit hierin zit, omdat dat het juist ingewikkeld maakt)